property_app/views.py

```diff
@@ -1,69 +1,181 @@
+
+        return Response(json_list)
     
     def delete(self, request, id):
-        pass
+        a_list = get_object_or_404(List_of_Properties, id=id)
+        a_list.delete()
+
+        return Response(status=HTTP_204_NO_CONTENT)
 
     def put(self, request, id):
-        pass
+        a_list = get_object_or_404(List_of_Properties, id=id)
+        json_list = ListSerializer(a_list, data=request.data, partial=True)
+
+        if json_list.is_valid():
+            json_list.save()
+            return Response(json_list.data, status=HTTP_204_NO_CONTENT)
+        else:
+            return Response(json_list.errors, status=HTTP_400_BAD_REQUEST)
+
+class Add_To_List(UserPermissions):
+    
+    def put(self, request, propid, listid):
+        a_property = get_object_or_404(Property, id=propid)
+        a_list = get_object_or_404(List_of_Properties, id=listid)
+
+        a_property.list_of_properties.add(a_list)
+        a_property.full_clean()
+        a_property.save()
+
+        json_property = PropertySerializer(a_property).data
+
+        return Response(json_property)
+
+class Remove_From_List(UserPermissions):
 
+    def put(self, request, propid, listid):
+        a_property = get_object_or_404(Property, id=propid)
+        a_list = get_object_or_404(List_of_Properties, id=listid)
 
+        a_property.list_of_properties.remove(a_list)
+        a_property.full_clean()
+        a_property.save()
+
+        json_property = PropertySerializer(a_property).data
+
+        return Response(json_property)
+
+class Add_To_Portfolio(UserPermissions):
+
+    def put(self, request, propid):
+        user, portfolio = get_user_and_portfolio(request)
+        
+        a_property = get_object_or_404(Property, id=propid)
+        a_property.portfolio = portfolio
+
+        a_property.full_clean()
+        a_property.save()
+
+        json_property = PropertySerializer(a_property).data
+
+        return Response(json_property)
+
+class Remove_From_Portfolio(UserPermissions):
+
+    def put(self, request, propid):
+        user, portfolio = get_user_and_portfolio(request)
+        
+        a_property = get_object_or_404(Property, id=propid)
+        a_property.portfolio = None
+
+        a_property.full_clean()
+        a_property.save()
+
+        json_property = PropertySerializer(a_property).data
+
+        return Response(json_property)
+
+class Purchase_Worksheet_View(UserPermissions):
+
+    def get(self, request, propid):
+        purchase_worksheet = Purchase_Worksheet.objects.get(matching_property=propid)
+        json_purchase_worksheet = PurchaseWorksheetSerializer(purchase_worksheet).data
+
+        return Response(json_purchase_worksheet)
+    
+    def put(self, request, propid):
+        purchase_worksheet = Purchase_Worksheet.objects.get(matching_property=propid)
+        json_purchase_worksheet = PurchaseWorksheetSerializer(purchase_worksheet, data=request.data, partial=True)
+
+        if json_purchase_worksheet.is_valid():
```

Comment:
It may be beneficial to handle requests that come in with invalid information and return a 400 status. Maybe a try and except of some form of error handling?